api/app/signals/apps/signals/management/commands/delete_signals.py

```diff
@@ -0,0 +1,143 @@
+
+        if options['state'] not in self.STATE_CHOICES:
+            raise ValueError(f'Invalid state(s) provided must be one of "{", ".join(self.STATE_CHOICES)}"')
+        self.state = options['state']
+
+        if options['days'] < 365:
+            raise ValueError('Invalid days provided must be at least 365')
+        self.days = options['days']
+
+        if options['signal_id'] and not Signal.objects.filter(id=options['signal_id'], parent_id__isnull=True).exists():
+            raise ValueError(f'Invalid signal_id provided, no (parent) Signal with id "{options["signal_id"]}" exists')
+        self.signal_id = options['signal_id'] or None
+
+        self._dry_run = options['_dry_run']
+        self.batch_uuid = uuid.uuid4()
+        self.now = timezone.now()
+
+    def handle(self, *args, **options):
+        try:
+            self._pre_handle(**options)
+        except ValueError as e:
+            self.stderr.write(f'{e}')
+            return
+
+        self.write_start_msg()
+
+        signal_qs = self._select_signals()
+        if signal_qs.exists():
+            with transaction.atomic():
+                self._loop_through_signals(signal_qs=signal_qs)
+        else:
+            self.stdout.write('No signal(s) found that matches criteria')
+
+        self.write_end_msg()
+
+    def _select_signals(self):
+        # Select Signals that are in the given states and the state has been set more than X days ago
+        # Only select normal signals or parent signals
+        queryset = Signal.objects.filter(parent_id__isnull=True, status__state=self.state,
+                                         status__created_at__lt=timezone.now() - timezone.timedelta(days=self.days))
+
+        if self.signal_id:
+            # A specific Signal ID was given, only select that Signal
+            queryset = queryset.filter(id=self.signal_id)
+
+        return queryset.filter()
+
+    def _loop_through_signals(self, signal_qs):
+        for signal in signal_qs:
+            if signal.is_parent:
+                # First delete all child Signals
+                self._loop_through_signals(signal_qs=signal.children.all())
+
+            # Delete the Signal
+            self._delete_signal(signal=signal)
+
+    def _delete_signal(self, signal):
+        """
+        Deletes the given signal and all related objects. Will also remove the Signal from the elasticsearch index.
+        """
+        signal_id = signal.id
+        if not self._dry_run:
+            # Meta data needed for reporting
+            deleted_signal = DeletedSignal.objects.create_from_signal(signal=signal, batch_uuid=self.batch_uuid)
+
+            # The log explaining how and why the signal was deleted
+            note = self.get_log_note(signal)
+            DeletedSignalLog.objects.create(deleted_signal=deleted_signal, action='automatic', note=note)
+
+            signal.attachments.all().delete()
```

Comment:
I dont think this wil delete any files from the storage as the delete is performed on the queryset and not the object itself.